### PR TITLE
COMMERCE-10847 Select order type before add product to a cart

### PR DESCRIPTION
**Pull request by jaydawu**
https://issues.liferay.com/browse/COMMERCE-8364
Resent from https://github.com/liferay-commerce/liferay-portal/pull/3780

### CODE REVIEW

**Comment by jaydawu** [comment]
ci:test:sf

**Comment by jaydawu** [comment]
ci:test:relevant

**Comment by jaydawu** [comment]
ci:test:commerce-functional

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceOrderTypes.testcase [hunk 1]
```diff
@@ -254,6 +383,338 @@ definition {
...
+		}
+
+		task ("Then the Select Order Type popup is shown again") {
+			AssertElementPresent(locator1 = "IFrame#MODAL_IFRAME");
+		}
+	}
+
+	@description = "This is a test for COMMERCE-11573. CanSelectOrderTypeWithEligibilityForANewOrder"
+	@priority = 3
+	test CanSelectOrderTypeWithEligibilityForANewOrder {
```
Comment (baharturk) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceOrderTypes.testcase:598]: CanSelectOnlyEligibleOrderType

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceOrderTypes.testcase [hunk 2]
```diff
@@ -254,6 +383,338 @@ definition {
...
+				password = "test",
+				urlAppend = "web/minium",
+				userEmailAddress = "buyer@liferay.com");
+
+			Click(
+				key_productName = "U-Joint",
+				locator1 = "CommerceAccelerators#ADD_TO_CART_BUTTON");
+		}
+
+		task ("Then the Select Order Type popup is shown with the two order types which setten for Minium Portal channel only") {
```
Comment (baharturk) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceOrderTypes.testcase:648]: "Then the Select Order Type popup is shown with the two order types eligible for Minium Portal channel"

**Review by baharturk — CHANGES_REQUESTED** [review]
Hi Jayda, please take a look at my comments. Thanks!

**Comment by jaydawu** [comment]
ci:test:sf

**Comment by jaydawu** [comment]
ci:test:commerce-functional

**Comment by jaydawu** [comment]
ci:test:relevant

**Review by baharturk — CHANGES_REQUESTED** [review]
Hi, please take a look at my comments. Thanks!

**Comment by jaydawu** [comment]
ci:test:sf

**Comment by jaydawu** [comment]
@baharturk could you review again? Thanks!

**Comment by jaydawu** [comment]
ci:test:relevant

**Comment by jaydawu** [comment]
ci:test:commerce-functional

File: modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceOrderTypes.testcase [hunk 3]
```diff
@@ -220,6 +220,225 @@ definition {
 		}
 	}
 
+	@description = "This is a test for COMMERCE-11560. Verify when we complete checkout and there is more than 1 active order type for the channel, the order type with the highest priority (lower alphanumeric) is selected on the Select Order Type popup"
```
Comment (baharturk) [inline modules/apps/commerce/commerce-product-test/src/testFunctional/tests/CPCommerceOrderTypes.testcase:223]: I guess you put the wrong description for this test, this description is used for the **CanViewHighestPriorityOrderTypeIsSetByDefault** test. 

Verify when we complete checkout and there is more than 1 active order type for the channel, the order type is the one selected on the Select Order Type popup.

**Review by baharturk — CHANGES_REQUESTED** [review]
Please take a look!

**Comment by jaydawu** [comment]
ci:test:sf

**Comment by jaydawu** [comment]
ci:test:sf

**Comment by jaydawu** [comment]
ci:test:sf

**Comment by jaydawu** [comment]
ci:test:commerce-functional

**Comment by jaydawu** [comment]
ci:test:relevant

**Comment by jaydawu** [comment]
ci:test:sf

**Comment by jaydawu** [comment]
ci:test:relevant

**Comment by jaydawu** [comment]
ci:test:commerce-functional

**Comment by jaydawu** [comment]
ci:test:commerce-functional

**Comment by jaydawu** [comment]
Hi @baharturk, this time should be good enough, please review, thank you!

**Comment by jaydawu** [comment]
ci:test:sf

**Comment by jaydawu** [comment]
ci:test:commerce-functional

**Comment by jaydawu** [comment]
ci:test:relevant

**Comment by baharturk** [comment]
ci:test:relevant

**Comment by jaydawu** [comment]
ci:test:relevant

**Comment by jaydawu** [comment]
ci:test:commerce-functional

**Comment by jaydawu** [comment]
All tests passed and the relevant errors are not related to this pr, it could be fast-forward, thanks!

cc/ @victoroscuro

**Comment by victoroscuro** [comment]
Based on the checks on the relevant and the commerce functional test suite, we can force forward this PR.

**Pull request closed without merging**